servicetalk-grpc-protoc/build.gradle

```diff
@@ -47,42 +50,78 @@ shadowJar {
   classifier = ''
 }
 
+def grpcPluginUberJarName = project.name + "-" + project.version + "-all.jar"
+
+task copyUberJarForDevelopment(type: Copy) {
+  dependsOn tasks.shadowJar
+  from shadowJar.outputs.files.singleFile
+  into file("$buildDir/buildExecutable")
+
+  rename { fileName ->
+    return grpcPluginUberJarName
+  }
+}
+
 task buildExecutable {
-  def outputFile = file("$buildDir/buildExecutable/protoc-gen-servicetalk_grpc.exe")
+  def isWindows = org.gradle.internal.os.OperatingSystem.current().isWindows()
+  def outputFile = new File("$buildDir/buildExecutable/" +
+      addExecutablePostFix("protoc-gen-servicetalk_grpc", isWindows))
+  dependsOn tasks.copyUberJarForDevelopment
   inputs.files shadowJar.outputs.files
   outputs.file outputFile
 
   doLast {
-    if (org.gradle.internal.os.OperatingSystem.current().isWindows()) {
-      throw new GradleException("Windows is currently not supported")
+    if (isWindows) {
+      buildWindowsExecutable(grpcPluginUberJarName, outputFile)
+    } else {
+      buildUnixExecutable(grpcPluginUberJarName, outputFile)
     }
-
-    outputFile.parentFile.mkdirs()
-    outputFile.delete()
-    outputFile <<
-        '''#!/bin/sh
-exec java -jar "$0" "$@"
-exit1
-'''
-    shadowJar.outputs.files.singleFile.withInputStream { outputFile.append(it) }
-    outputFile.executable = true
   }
 }
 tasks.compileJava.finalizedBy(buildExecutable)
 
+task buildExecutableWindowsPublishing {
+  def outputFile = new File("$buildDir/buildExecutable/" +
+      addExecutablePostFix("protoc-gen-servicetalk-windows_grpc", true))
+  dependsOn tasks.copyUberJarForDevelopment
+  inputs.files shadowJar.outputs.files
+  outputs.file outputFile
+
+  doLast {
+    buildWindowsExecutable(grpcPluginUberJarName, outputFile)
+  }
+}
+
+// we attempt to generate both grpc executables when on windows, and don't publish from windows anyways.
+tasks.withType(PublishToMavenRepository) {
```

Comment:
`PublishToMavenRepository` does not cover `PublishToMavenLocal`. Should we use `AbstractPublishToMaven` type instead?
https://docs.gradle.org/current/javadoc/org/gradle/api/publish/maven/tasks/AbstractPublishToMaven.html

Reply:
I actually want `PublishToMavenLocal` to work for local testing. This is just meant to highlight the fact that we can generate all artifacts from a single architecture (e.g. just a jar and a script for windows). It could be removed but the hope is that it helps clarify expectations.